python/pyspark/pandas/tests/test_generic_functions.py

```diff
@@ -200,6 +200,22 @@ def test_stat_functions(self):
         self.assert_eq(pdf.b.kurtosis(), psdf.b.kurtosis())
         self.assert_eq(pdf.c.kurtosis(), psdf.c.kurtosis())
 
+    def test_prod_precision(self):
```

Comment:
this test will fail in original implementation due to precision loss

the new implementation will provide the precise result for integral inputs, when the product in [Long.MinValue, Long.MaxValue]